### DISCUSSION TITLE
[Image] Add custom transformation parameters for external loaders

### DISCUSSION THREAD

**Discussion by arguiot** [vercel/next.js#19852] [Ideas]
# Feature request
It would be great to overwrite/add transformation parameters for external loaders. 

Currently, when adding URL parameters in the src URL, NextJS ignores them. So you end up having 2 `?` symbols in your URL which results in a 404 error.

## Is your feature request related to a problem? Please describe.
I use Imgix for serving images, and not having access to transformation parameters is a real pain.

## Describe the solution you'd like
For example, when using Imgix, you could write something like:
```jsx
<Image src="img.jpeg" width={600} height={600} transformations={{ 
    auto: "compress",
    fit: "crop",
    mask: "ellipse",
    fm: "png"
}} />
```
And it would apply the transformation parameters for each URL in the srcset.

## Describe alternatives you've considered
Using the standard `<img>` tag. But it's not ideal...

**Comment by allanlasser** [comment]
I've encountered a similar issue. The `crop=faces` argument is something we use a lot, since it helps avoid awkward crops on our key images, which are used in multiple aspect ratios (1:1, 2:3, 120:63, etc.).

@arguiot, since you originally said:

> Currently, when adding URL parameters in the src URL, NextJS ignores them. So you end up having 2 ? symbols in your URL which results in a 404 error.

This makes me think that an alternative solution could be to merge any query arguments provided on the src string with the query arguments that Next wants to append.

**Update**

Looked into it, and it seems like a straightforward fix for this would be in [the `imgixLoader` function](https://github.com/vercel/next.js/blob/a03f1f3847b827ff1ae236111267cdf81ef9ba75/packages/next/client/image.tsx#L417-L429). We could look for an existing `paramsString` on `src` instead of assuming it's empty, then append the joined params to either the existing `paramsString` or `'?'`.

**Reply by arguiot** [reply]
That could be a great solution. Maybe you should open a pull request!
